### PR TITLE
bug 9765; fix citation tree view for HandleError exceptions

### PR DESCRIPTION
**Pull request by prculley**
A relatively recent change makes attempts to look up objects via db.get_*_from_handle(), when the handle is not present in the db, cause an exception, where it used to just return a None.

The citation tree view has both sources and citations in the same list; so when a list line is selected for an operation, the code has to try both sources and citations to figure out which is actually selected.  When the code attempted to lookup both source and citation with the same handle, one always failed with an exception.

This PR corrects this by using try/except syntax in place of the None tests.

A secondary bug found was that following a merge, the tree view was not updated and an attempt to select and do something one of the 'merged out' entries would fail.  I forced a view update following a merge.

### CODE REVIEW

File: gramps/plugins/view/citationtreeview.py [hunk 1]
```diff
@@ -555,6 +541,23 @@ def get_handle_from_gramps_id(self, gid):
         else:
             return None
 
+    def get_source_or_citation(self, handle, check=True):
```
Comment (QuLogic) [inline gramps/plugins/view/citationtreeview.py]: Should it be private?

**Comment by sam-m888** [comment]
Do you need to respond to QuLogic's review, before this can be accepted?

**Comment by prculley** [comment]
I had not intended to respond, Gramps code has so many methods that -should- be private, but are not, that I didn't think we were particularly concerned about this.  I actually don't think about it at all during coding.

If, as a policy, we are going to start enforcing this, then I think we have some real work in going through all the code to get it to comply.

**Comment by QuLogic** [comment]
Marking one _new_ method as private is not the same thing as enforcing it across the code base.

But even a "no" is better than silence...

**Comment by sam-m888** [comment]
Thank you

**Pull request merged**